split parse.py to modules - phase 1

agvania opened the pull request:
The idea - to separate parse.py to (at least) three parts:
1. the parser - takes docx and transforms it to an abstract document, as was defined by Zvika (meaning every text segment has its own type, which is defined based on the way that text is displayed by the Rav in the Milon.
2. Document Builder(s) - taking an abstract document and builds, according to the types defined in it for every text segment, the actual document. This actual document can HTML, LaTex, or whatever.
3. The module that connects between the parser and the builder. This module can have more than one builder, so that we can create several documents from the same abstract document.
As a start, take the translation of docx codes to text segment types out
of parse.py, in order to reduce dependencies and improve readability.

agvania committed:
split parse.py to modules - phase 1

As a start, take the translation of docx codes to text segment types out
of parse.py, in order to reduce dependencies and improve readability.

agvania committed:
cont. phase 1

Added segment types and updated parse.py accordingly.

ZvikaZ commented:
I think I will pull this when it will be more complete.
Currently, I don't see "the whole picture" yet...

BTW,
I think if you're at splitting, we need also to split out all the controlling fields (`process, create_*, doc_file_name`) and the executing code at the end of `parse.py` (`if create*`, and downward) to a seperate file. Maybe `main.py`.

If I will be in the mood, maybe I will write later a simple GUI that will call `main.py`'s controlling function with appropriate values.

agvania committed:
Merge remote-tracking branch 'refs/remotes/ZvikaZ/master' into split_parse

agvania committed:
separate fixes and building html doc from the main script.

We take the parts of the code that are responsible for building the html
documents into separate modules.

agvania committed:
separate parsing and building from main script

created build_* modules and docxparser module, to separate the processes
from each other, and eliminate the dependency between them.
MEANING - no more references to html_doc from code that is responsible
for parsing, and no more if-else blocks checking for latex or html build
in the parsing process, etc.

agvania committed:
separate parse.py - final phase

parse.py changed to main.py

We have docxparser.py - the module transforming docx paragraphs into
abstract paragraphs
html_build.py- module for building the html docs from the abstract
paragraphs
latex_build.py - module for building the .tex file from the abstract
paragraphs.
multi_build.py - aggregates all the builders. Thus we can have several
builds at once.

The general pipeline of the milon is this:
docx --> parser --> abstract --> builders

agvania committed:
Builders, zipper, and phone-gap

Sharpened builders, and created multi_build, for abstraction of the
multiple builders (html and latex) to one builder.
created zipper, which decorates the html builder, meaning it does
exactly the same, only it zips the milon at the end.
created PhoneGap builder, which decorates the zipper, meaning it creates
the .apk at the end.

Also, now with the decorators, it is easy to decide which actions we
want to perform in the pipeline. Just comment out the parts that are
responsible for those actions.

agvania committed:
updated comment

agvania commented:
I have finished all the splitting process, with success.
ran with dict_check.docx and with dict.docx, and it looks the same in the browser.
one thing that is changed, and I don't know why, is that the html docs names, which are printed in the end of the creation, are not in the same as it was before the split. Yet. this change does not effect the final output of the milon is the same.
Since the dict_check.docx file was changed by Zvika, I recommend that the new script will be run on the new dict_check.docx to make sure it passes the test.

agvania committed:
updated comment

ZvikaZ commented:
Sorry for the too long delay.
(I really was on reserve duty, part of the time :)  )

Great job!

But, on first observation - there are 2 problems:
- Latex creation doesn't work - gives `global name 'TS' is not defined`
- Have you diff'ed `debug_fix.txt`? 
  Maybe I forgot to tell you, but that's the important file to diff between `output` and `output golden`.
  It contains many changes in that pull version.

I've pushed my code, there were very few changes in it in the last month - but in contains the updated dict_check.docx for sure.

And maybe we should try to understand what happened to the file names - maybe there is some 'dragon' hiding behind this...

Thanks again, and sorry again...

agvania committed:
Merge remote-tracking branch 'refs/remotes/ZvikaZ/master' into split_parse

agvania commented:
I'm trying to run parse.py on **your** branch, but I get

Enter file name: 

and when I try to enter a name, I get no response.
How should I run it?

regarding the debug.txt file - after I'll solve the latex problem I'll get to it.

agvania committed:
latex_check

agvania committed:
fixes for latex

agvania committed:
minor fixes of debug output
